PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterCharacterScripts/Sound.rbxmx

```diff
@@ -0,0 +1,381 @@
+	Head = Figure:WaitForChild("Head")
+	Humanoid = Figure:WaitForChild("Humanoid")
+	
+	Sounds[SFX.Died] = 			Head:WaitForChild("Died")
+	Sounds[SFX.Running] = 		Head:WaitForChild("Running")
+	Sounds[SFX.Swimming] = 		Head:WaitForChild("Swimming")
+	Sounds[SFX.Climbing] = 		Head:WaitForChild("Climbing")
+	Sounds[SFX.Jumping] = 		Head:WaitForChild("Jumping")
+	Sounds[SFX.GettingUp] = 	Head:WaitForChild("GettingUp")
+	Sounds[SFX.FreeFalling] = 	Head:WaitForChild("FreeFalling")
+	Sounds[SFX.Landing] = 		Head:WaitForChild("Landing")
+	Sounds[SFX.Splash] = 		Head:WaitForChild("Splash")
+end
+
+local Util
+Util = {
+	
+	--Define linear relationship between (pt1x,pt2x) and (pt2x,pt2y). Evaluate this like at x.
+	YForLineGivenXAndTwoPts = function(x,pt1x,pt1y,pt2x,pt2y)
+		--(y - y1)/(x - x1) = m
+		local m = (pt1y - pt2y) / (pt1x - pt2x)
+		--float b = pt1.y - m * pt1.x;
+		local b = (pt1y - m * pt1x)
+		return m * x + b
+	end;
+	
+	--Clamps the value of "val" between the "min" and "max"
+	Clamp = function(val,min,max)
+		return math.min(max,math.max(min,val))	
+	end;
+	
+	--Gets the horizontal (x,z) velocity magnitude of the given part
+	HorizontalVelocityMagnitude = function(Head)
+		local hVel = Head.Velocity + Vector3.new(0,-Head.Velocity.Y,0)
+		return hVel.magnitude	
+	end;
+	
+	--Gets the vertical (y) velocity magnitude of the given part
+	VerticalVelocityMagnitude = function(Head)
+		return math.abs(Head.Velocity.Y)
+	end;
+	
+	--Checks if Sound.Playing is usable
+	SoundIsV2 = function(sound)
+		return sound.IsLoaded
+	end;
+	
+	--Setting Playing/TimePosition values directly result in less network traffic than Play/Pause/Resume/Stop
+	--If these properties are enabled, use them.
+	Play = function(sound)		
+		if Util.SoundIsV2(sound) then
+			if sound.TimePosition ~= 0 then
+				sound.TimePosition = 0
+			end
+			if not sound.IsPlaying then
+				sound.Playing = true
+			end
+		else
+			sound:Play()
+		end
+		
+	end;
+	Pause = function(sound)
+		if Util.SoundIsV2(sound) then
+			if sound.IsPlaying then
+				sound.Playing = false
+			end
+		else
+			sound:Pause()
+		end
+	end;
+	Resume = function(sound)
+		if Util.SoundIsV2(sound) then
+			if not sound.IsPlaying then
+				sound.Playing = true
+			end
+		else
+			sound:Resume()
+		end
+	end;
+	Stop = function(sound)
+		if Util.SoundIsV2(sound) then
+			if sound.IsPlaying then
+				sound.Playing = false
+			end
+			if sound.TimePosition ~= 0 then
+				sound.TimePosition = 0
+			end
+		else
+			sound:Stop()
+		end
+	end;
+}
+
+do
+	-- List of all active Looped sounds
+	local activeLoopedSounds = {}
+	
+	-- Last seen Enum.HumanoidStateType
+	local activeState = nil
+	
+	-- Verify and set that the sound's .Looped property is the value of "looped"
+	function verifyAndSetLoopedForSound(sound, looped)
+		if sound.Looped ~= looped then
+			sound.Looped = looped
+		end
+	end
+	
+	-- Verify and set that "sound" is in "activeLoopedSounds".
+	function setSoundInActiveLooped(sound)
```

Comment:
this is a little different:
so this localscript maintains a list of "looped sounds" that are actively playing.
this "setSoundInActiveLooped" adds the parameter sound to the "activeLoopedSounds" array.